gravitee-am-service/src/test/java/io/gravitee/am/service/validators/FlowValidatorImplTest.java

```diff
@@ -0,0 +1,149 @@
+/**
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.am.service.validators;
+
+import io.gravitee.am.model.flow.Step;
+import io.gravitee.am.service.exception.InvalidParameterException;
+import io.gravitee.am.service.model.Flow;
+import io.gravitee.am.service.validators.flow.FlowValidatorImpl;
+import io.gravitee.am.service.validators.flow.policy.SendEmailPolicyValidator;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+
+import static org.mockito.AdditionalMatchers.not;
+import static org.mockito.Mockito.*;
+
+@ExtendWith(MockitoExtension.class)
+public class FlowValidatorImplTest {
+
+    @Mock
+    private SendEmailPolicyValidator emailPolicyValidator;
+
+    @InjectMocks
+    private FlowValidatorImpl flowValidator;
+
+    @Test
+    void must_invalidate_none() {
+        Step step1 = getStep("step1", "{\"policy\":\"policy1\"}");
+        Step step2 = getStep("step2", "{\"policy\":\"policy2\"}");
+        Flow flow = new Flow();
+        flow.setName("testFlow");
+        flow.setPre(List.of(step2));
+        flow.setPost(List.of(step1));
+
+        when(emailPolicyValidator.validate(any())).thenReturn(true);
+
+        var observer = flowValidator.validate(flow).test();
+        observer.awaitDone(10, TimeUnit.SECONDS);
+        observer.assertComplete().assertNoErrors();
+
+        verify(emailPolicyValidator, times(2)).validate(any());
+    }
+
+    @ParameterizedTest
+    @MethodSource("params_that_must_invalidate_step")
+    void must_invalidate_step(String invalidStepName, int expectedInvocations) {
```

Comment:
Can variables declaration be adjusted to have consistency? In the same method there are variables declared with `final`, some of them without it and other with `var`.
In general I prefer to use `var` and leave the compiler to do the work for me. But that's just personal preference 😄

Reply:
Will do